Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,75 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries is selfintersecting. And have been disregarded.");
+            }
+
+            // Runs Boolean Union to resolve intersections and containemnt
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || x is Ellipse)
+                                .Concat(internalBoundaries.Where(x => !(x is NurbsCurve) && !(x is Ellipse)).BooleanUnion()).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries was overlapping another one. BooleanUnion was used to resolve it.");
+            }
+
+            //// Checks based around external curve
+            if (externalBoundary is NurbsCurve)
+            {
+                Reflection.Compute.RecordWarning("Neeseccary checks to ensure vadility of the PlanarSurface is not implemented. The PlanarSurfaces curves relations are not garanteueed.");
+                // External done
+                return new PlanarSurface(externalBoundary, internalBoundaries);
+            }
+            else if (externalBoundary.IIsSelfIntersecting())
+            {
+                Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                return null;
+            }
+            if (externalBoundary is Ellipse)
+                return new oM.Geometry.PlanarSurface(externalBoundary, internalBoundaries);
+
+
+            // Intersects External
+            for (int i = 0; i < internalBoundaries.Count; i++)
+            {
+                ICurve intCurve = internalBoundaries[i];
+                if (intCurve is NurbsCurve || intCurve is Ellipse)
+                    continue;
+
+                if (externalBoundary.ICurveIntersections(intCurve).Count != 0)
+                {
+                    externalBoundary = externalBoundary.BooleanDifference(new List<ICurve>() { intCurve }).Single();
+                    internalBoundaries.RemoveAt(i);
+                    i--;
+                    Reflection.Compute.RecordWarning("At least one of the internalBoundaries is intersecting the externalBoundary. BooleanDifference was used to resolve the issue.");
+                }
+            }
+
+            // IsContained by External
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || x is Ellipse || externalBoundary.IIsContaining(x)).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries is not contained by the externalBoundary. And have been disregarded.");
+            }
```

Comment:
Why is this happening so late?

Reply:
If a internalCurve intersects the external one it's not contained and needs to be handled by boolean difference, which is what happens before this.

Both of these can only happen if the external curve isn't a NurbsCurve/Ellipse, which happens just before that.

Reply:
Oke, I trust you are right (cannot find a valid argument against)